lib/rbkit/server.rb

```diff
@@ -7,7 +7,7 @@ def initialize(pub_port, request_port)
       @request_port = request_port.to_i
       [@pub_port, @request_port].each{|port| validate_port_range(port) }
       @profiler_thread = nil
-      @profiler_stop_thread = false
+      @stop_profiler_thread = false
```

Comment:
>  It segfaults only when runnning the benchmark program.

I think we need to investigate this more then. I thought we figured out the root cause which was that the message dispatch thread doesn't terminate after stopping the server due the wrong variable name of "stop_profiler_thread". Looks like something else is happening ?

> we can't quite test for a segmentation fault isn't it?
The test itself would segfault and come out with an exit status indicating failure.

```ruby
it "does not segfault"
```
Not a great way to test, but atleast the segfault is covered in the test.